draft-ietf-core-corr-clar.md

```diff
@@ -535,6 +535,28 @@ Content-Format registrations.
 These are not repeated here; they are however quite useful as a
 reference when preparing additional Content-Format registrations.
 
+## RFC 7252-5.10.2: Proxy-Scheme
+
+RFC 7252 differentiates between requests made to a forward-proxy
+that contain the Proxy-Scheme option (or the Proxy-Uri option)
+and regular requests where no Proxy- option is set.
+
+For a unified view on requests, it can be beneficial for implementers (especially of servers)
+to treat Proxy-Scheme the same way as Uri-Host
+in that it expresses a part of the request URI,
+and in that it has a default value influenced by the scheme indicated in the transport's URI compositon rules.
```

Comment:
The term "default" really should be used for CoAP's Opton processing only.
With a "default", proxy-scheme would always have a value and Proxy-Scheme-Number could not be introduced.

Reply:
Having a "default" helps senders and and recipients agree on a value, and helps emphasize that conceptually it is always there.

Having alternative expressions for that override does indeed make things complicated. Maybe I can come up with text that is more "Conceptually it's always there so it doesn't really do anything to set it", which would also have the nice effect of being less contradicting/updating and more explaining.

Reply:
So this really should be text that applies when neither Proxy-Scheme nor Proxy-Scheme-Number are present.
We could also put this text into href, so we don't have to put anything here.

Reply:
How can text in href influence CoAP implementations? While it's a useful tool to avoid decoding and reencoding, normatively it's just a different way of expressing a URI (reference).

Reply:
> How can text in href influence CoAP implementations?

By defining the CoAP Option [Proxy-Scheme-Number][] ....

[1]: https://www.ietf.org/archive/id/draft-ietf-core-href-22.html#name-proxy-scheme-number

Reply:
That would make Proxy-Scheme-Number not be a 1:1 analog for Proxy-Scheme … but if we're willing to go there (I am), yes, that could state that a present Proxy-Scheme-Number option that matches the default scheme of the transport is equivalent to its absence. That'd effectively force implementations that use it to apply the same reasoning to Proxy-Scheme as well.

If this were any other WG's document I'd protest that a new option only works out when taking a particular stance, but as href is a CoRE document, why not.